src/shimoku_api_python/api/plot_api.py

```diff
@@ -651,6 +636,79 @@ def _check_mapping_in_report_data_set(self, report_data_set: Report.ReportDataSe
             logger.warning(f'Mapping inconsistent in component data set link {str(report_data_set)}')
         return report_data_set_mapping == mapping
 
+    @async_auto_call_manager()
+    @logging_before_and_after(logging_level=logger.debug)
+    async def filter(
+        self, order: int, data: str, field: str, multi_select: bool = False,
+        cols_size: int = 4,  rows_size: int = 1, padding: Optional[str] = None
+    ):
+        """ Filter the data set.
+        :param order: the order of the chart
+        :param data: the data
+        :param field: the field to filter
+        :param multi_select: whether to allow multi select
+        :param cols_size: the size of the columns
+        :param rows_size: the size of the rows
+        :param padding: the padding
+        """
+
+        if not isinstance(data, str):
+            log_error(logger, f'To filter a data set, the data must be a shared data entry', DataError)
+
+        if data not in self._shared_data_map:
+            log_error(logger, f'No shared data with name {data} found', DataError)
+
+        _, data_set, _ = self._shared_data_map[data][field]
+        data = self._shared_data[data]
+
+        data_columns = data.columns.to_list()
+        converted_data = pd.DataFrame(convert_input_data_to_db_items(data.to_dict(orient='records')))
+        converted_data_columns = converted_data.columns.to_list()
+
+        series_name = converted_data_columns[data_columns.index(field)]
+        options = None
+        operations = None
+        if series_name.startswith('date'):
+            input_type = "DATERANGE"
+            operations = ['contains']
+        elif series_name.startswith('int'):
+            input_type = "NUMERIC"
+        elif series_name.startswith('string'):
+            operations = ['eq']
+            if multi_select:
+                input_type = "CATEGORICAL_MULTI"
```

Comment:
This could be typed via enum:

```python
from enum import Enum, auto

class InputType(Enum):
    DATERANGE = "DATERANGE"
    NUMERIC = "NUMERIC"
    CATEGORICAL_SINGLE = "CATEGORICAL_SINGLE"
    CATEGORICAL_MULTI = "CATEGORICAL_MULTI"
```

and then you can access via:

InputType.DATERANGE etc

Reply:
In this case it is only used with the filter thats why i didnt extract the code, but its not a problem to implement it with an enumerator, this way the LSP will be able to recognise it